lucille-core/src/main/java/com/kmwllc/lucille/connector/FileConnector.java

```diff
@@ -0,0 +1,182 @@
+package com.kmwllc.lucille.connector;
+
+import com.kmwllc.lucille.connector.cloudstorageclients.CloudStorageClient;
+import com.kmwllc.lucille.core.ConnectorException;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Publisher;
+import com.typesafe.config.Config;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Cloud Options based on providers:
+ *  Google:
+ *    "pathToServiceKey" : "path/To/Service/Key.json"
+ *  Azure:
+ *    connectionString" : azure connection string
+ *      Or:
+ *    "accountName" : azure account name
+ *    "accountKey" : azure account key
+ *  Amazon:
+ *    "accessKeyId" : s3 key id
+ *    "secretAccessKey" : secret access key
+ *    "region" : s3 storage region
+ *
+ * Optional:
+ *  "maxNumOfPages" : number of reference to the files loaded into memory in a single fetch request, defaults to 100
+ */
+
```

Comment:
extra space

Reply:
removed in [1082fb3](https://github.com/kmwtechnology/lucille/pull/205/commits/1082fb30901ad893f413e5243c62ff2ec98b48d5)